server/src/main/java/org/eclipse/lsp/cobol/service/utils/WorkspaceFileService.java

```diff
@@ -60,12 +60,13 @@ public String decodeURI(@NonNull String uri) {
   @Nullable
   @Override
   public String getNameFromURI(@NonNull String uri) {
-    try {
-      return FilenameUtils.getBaseName(Paths.get(new URI(uri)).getFileName().toString());
-    } catch (URISyntaxException e) {
-      LOG.error("Cannot get file name from: " + uri, e);
-      return null;
-    }
+    return new File(uri).getName().replaceFirst("\\..*$", "");
```

Comment:
This will not work with dot-separated dataset names: like `FO2345.COBOL(PROG).CBL`.